### DISCUSSION TITLE
Updating 3.3.1 to newer versions fail. glibc issue?

### DISCUSSION THREAD

**Discussion by adrian-schutte** [zwave-js/node-red-contrib-zwave-js#61] [Q&A] [answered]
Good day.

I have been getting errors when trying to update from 3.3.1 to the newer versions. I tried to update to 3.7.1 now and still having issues.

It looks like the update wants glibc ver 2.28 but my distro, opensuse leap 15.2, is on 2.26. Is there a way to get past this step with glibc ver 2.26?

Here is the latest error log. [2021-05-31T10_12_34_850Z-debug.log](https://github.com/zwave-js/node-red-contrib-zwave-js/files/6569640/2021-05-31T10_12_34_850Z-debug.log)

I hope this is the right place to ask for help. This is the first time I am interacting on github.

**Comment by marcus-j-davies** [comment, marked as answer]
Hi,

Is this being run under Home Assistant?
or is this a standalone Node-Red install?

How is the process being launched, I note its under root. is that via pm2 or a systemd service?

**Reply by adrian-schutte** [reply]
Hi,

Standalone Node-Red install.  Via pm2.

**Reply by marcus-j-davies** [reply]
Ok - I use pm2 and  when it runs NPM (in the root home dir), NPM doesn't see sudo (no idea why)
whenever `serialport` needs to be recompiled, It never works with PM2 (at least in the root dir)

Can you:
1 Stop the PM2 Node-Red instance
2 enter into a sudo context `sudo su`
3 cd to `/root/.node-red/`
4 install manually (with sudo) `sudo npm install node-red-contrib-zwave-js@latest`
5 Start  the PM2 Node-Red instance

and let me know if that worked

**Reply by adrian-schutte** [reply]
I am logged in as root so strictly don't need to sudo. I tried your steps but it still does not work. Here is the latest log.

[2021-05-31T13_25_33_271Z-debug.log](https://github.com/zwave-js/node-red-contrib-zwave-js/files/6570553/2021-05-31T13_25_33_271Z-debug.log)

**Reply by marcus-j-davies** [reply]
Ok,

Look like the `seriaports` package have increased some min requirements (or one of its dependencies).

as a temp work around.
You could clone the package, revert to a `serialports` package that was working for your envinement, then install.


`cd /root/.node-red/`
`git clone https://github.com/zwave-js/node-red-contrib-zwave-js.git`
`nano ./node-red-contrib-zwave-js/package.json`

Change     `"serialport": "9.1.0"` to     `"serialport": "9.0.7"` then save

`sudo npm install ./node-red-contrib-zwave-js`

This will get 3.7.1 installed with the serialports version used in 3.3.1.

I will  look into this.

Let me know either way.

**Reply by marcus-j-davies** [reply]
Another solution is to install glibc 2.28 via your package manager, if its exists in the repository.
but there is a risk, bad things could happen - so be cautious, if you try this.

**Reply by adrian-schutte** [reply]
The workaround works.

glibc 2.28 is not available in the opensuse repo's.

**Reply by marcus-j-davies** [reply]
Ok - I'm glad you managed to update.

glibc 2.28 was released in 2018 - so I'm supprised the repo / image has not moved across (I'm not overly experienced in the *nix fundamentals - so I'm sure there is a valid reason).

Installing a npm package from a folder, creates a symlink to it (in the node_nodules dir), so dont delete the cloned directory.

**Reply by adrian-schutte** [reply]
OK.

Thank you for the assistance.

**Reply by marcus-j-davies** [reply]
Your Welcome
